### ISSUE TITLE
Python Kernel Crashes Upon Importing Polars

### ISSUE THREAD

**Issue by fabianjkrueger** [getcursor/cursor#1947]
Upon importing polars (https://docs.pola.rs/), the kernel crashes. This does not happen in VS Code, even when I use the exact same environment.

### Screenshot
![Screenshot 2024-09-20 at 21 06 39](https://github.com/user-attachments/assets/44017010-3e38-40ce-a2fe-8b8176590506)

### Operating System
Running Cursor on macOS Sonoma 14.3.1
Connected to Ubuntu 22.04.4 LTS via SSH to run the code

### Steps to Reproduce
1. Create new conda environment
2. Activate that conda environment
3. Install polars library. (I did it like this: `pip install polars`, and afterwards also like this`conda install polars`. Both led to bug.)
4. Start interactive Jupyter session
5. Import polars

**Comment by danperks** [comment]
Did you see what the errors were in the Jupyter log? Unless this is caused by a bug fixed in newer VSCode releases that aren't available in Cursor yet, I'd be surprised if it was a Cursor bug.

**Comment by fabianjkrueger** [comment]
Hi. 
I found a solution / workaround and wanted to share it in case someone else runs into the same problem in the future.

### TLDR
```
# remove your current polars installation
conda remove polars
# or
pip uninstall polars

# install polars long term support for CPU in its place
pip install polars-lts-cpu
```
This solves the issue for me, and I can use polars in cursor.

### Bug Report

After spending some time working in vs code because of this, I decided to give cursor another try, but for my current data, pandas just takes too long without setting up multi core processing.
The issue is still the same: The jupyter kernel crashes upon importing polars.
It's even more than that: It was neither possible to run scripts depending on polars, nor to start a python session in the command line and import polars there. Basically it was not usable for me from within cursor at all.

These are the error logs when polars is imported in a python session in the command line:
```
RuntimeWarning: Missing required CPU features.

The following required CPU features were not detected:
    avx2, fma, bmi1, bmi2, lzcnt, movbe
Continuing to use this version of Polars on this processor will likely result in a crash.
Install the `polars-lts-cpu` package instead of `polars` to run Polars with better compatibility.

Hint: If you are on an Apple ARM machine (e.g. M1) this is likely due to running Python under Rosetta.
It is recommended to install a native version of Python that does not run under Rosetta x86-64 emulation.

If you believe this warning to be a false positive, you can set the `POLARS_SKIP_CPU_CHECK` environment variable to bypass this check.

  warnings.warn(
Illegal instruction (core dumped)
```

The CPU features `avx2, fma, bmi1, bmi2, lzcnt, movbe` were introduced around 2013-2015 for performance optimization of x86-64 processors.

I opened a regular CLI terminal (bash) and checked if they are indeed not installed:
```
grep -m1 flags /proc/cpuinfo | grep -o -E 'avx2|fma|bmi1|bmi2|lzcnt|movbe' 
fma
movbe
bmi1
avx2
bmi2
```

So yeah, `lzcnt` is actually missing apparently. Still, in vs code or a regular python session in a command line opened via SSH connecting to the server using my native terminal, it's still possible to import polars without an error.

It's interesting, because the exact same code in the exact same conda environment ran on the exact same machine works when executed from within vs code. Nothing is changed except the IDE. Because of that, @danperks I do think the error has to to something with cursor at least. For some reason, it seems like cursor cannot access the CPU features.
